tests/ssl-opt.sh

```diff
@@ -12394,12 +12456,14 @@ run_test    "TLS 1.3: Server side check - mbedtls with sni" \
             -s "parse ServerName extension" \
             -s "HTTP/1.0 200 OK"
 
-for i in opt-testcases/*.sh
-do
-    TEST_SUITE_NAME=${i##*/}
-    TEST_SUITE_NAME=${TEST_SUITE_NAME%.*}
-    . "$i"
-done
+if [ "$LIST_TESTS" -eq 0 ]; then
```

Comment:
Surely the inclusion of the other files still needs to be done when listing test cases? How else would we list the test cases in the other files?

Reply:
From what I understand, I am reproducing the same functionality that was in place before:
```
        with open(file_name, 'rb') as file_contents:
            for line_number, line in enumerate(file_contents, 1):
                # Assume that all run_test calls have the same simple form
                # with the test description entirely on the same line as the
                # function name.
                m = re.match(br'\s*run_test\s+"((?:[^\\"]|\\.)*)"', line)
                if not m:
                    continue
                description = m.group(1)
                self.process_test_case(descriptions,
                                       file_name, line_number, description)
```

This opens the ssl-opt.sh file, goes line by line finding the "run_test" occurrences and then adds the description, so no other files (and no other run_test that are not from the ssl-opt.sh file) are involved.
I can delete this and include the rest of the files, but I was trying to reproduce the same functionality as before (except of course for the $1 parameter replacement), and this was verified as indicated in the previous comments

Reply:
That chunk of code was executed [not only over `ssl-opt.sh` but also over `opt-testcases/*.sh`](https://github.com/Mbed-TLS/mbedtls/pull/8088/files#diff-593cad7aeb616419d1cc514afc2565034226081397deefd3fd77a6a873a66ec4L136).

(And even if it wasn't, that would have been a bug that we'd be fixing with the new approach, like we're fixing the bug that test cases where the call to `run_test` uses variable interpolation in the first argument weren't analyzed properly.)

Reply:
I "reverted" the change

Reply:
This is not resolved as of 52e36e93a8408f8663aeb558211cde847ba974c1, so the test cases from `tests/opt-testcases/*` are missing from the output of `ssl-opt.sh --list-test-cases`.